Firestore/src/FirestoreSessionHandler.php

```diff
@@ -396,6 +396,8 @@ private function commitTransaction()
 
             throw $e;
         }
+        // Begin a new transaction.
+        $this->open($this->savePath, $this->sessionName);
```

Comment:
Can you add some color to the reasoning here? My understanding is that between the various operations (reads and writes, deletes and gc), we could be prone to trying to write on a closed transaction. To this problem I should add another: It is considered incorrect to open a transaction without a corresponding rollback or commit. [0]

Given these two considerations, it seems to me that the best solution to this would be to defer commit until the handler calls `close()`. That way we don't have calls to closed transactions and everything will be committed or rolled back.

Of course, always glad to hear alternatives! :)

[0] From the Firestore Server API design doc:

> CommitRequest is used to persist all writes in this transaction and to release all locks.. No more operations can occur after a commit. To clear up server state, commit must be called even when they are no pending writes.
> RollbackRequest is used to release all locks without sending any writes.

Reply:
Done! GC now runs in its own transaction, everything else gets commited on `close`